### ISSUE TITLE
Add edit habit page 

### ISSUE THREAD

**Issue by FelixJncs** [FelixJncs/capstone-project#4]
## Value proposition

As a user 
I want to edit my habit data
in order to modify the habit name, reason, feeling and how to overcome 

## Description
![EditHabit Page](https://user-images.githubusercontent.com/125086329/236433719-aebfe6db-9d32-4c9b-9c9a-4ea48b2e3b45.svg)

## Acceptance criteria

- [x] The Habit List page displays a list of habits with an Edit button next to each habit
- [x] Clicking the Edit button for a habit takes the user to the EditHabitPage component for that habit
- [x] The EditHabitPage component renders a form with pre-filled inputs for habit name, reason, feeling and how to overcome
- [x] User can modify the pre-filled inputs for habit name, reason, feeling and how to overcome
- [x] User can save the changes to the habit data, which updates the habit record in the database
- [x] User can cancel the editing process, which redirects them back to the previous page without saving any changes.

## Tasks

- [x] Add an Edit button to each habit in the Habit List page
- [x] Create component/ EditHabitPage
- [x] Retrieve the current habit data from the database and pre-fill the input fields with that data
- [x] Update the habit record in the database when the user saves changes
- [x] Redirect the user back to the Habit List page when they cancel the editing process

Size: L

**Comment by ahohnsen** [comment]
Hi Felix,

Great job 🚀 

I have a few minor suggestions:

- Value proposition: I would rephrase the last part. Think about, why a user would like to edit the habit information: cause he/she maybe made a mistake or wants to keep the information up-to-date. My suggestions would be something like:

_As a user
I want to edit my habit data
in order to keep my habit data up-to-date._

- Acceptance criteria: _User can save the changes to the habit data, which updates the habit record in the database_ 👈🏼 Maybe you can add here, what happens after saving. Is there a redirection?
